build/src/rust/parser.rs

```diff
@@ -44,29 +46,28 @@ pub async fn generate_java(repo_root: &RepoRoot) -> Result {
 pub async fn run_self_tests(repo_root: &RepoRoot) -> Result {
     let base = &repo_root.target.generated_java;
     let lib = &repo_root.lib.rust.parser.generate_java.java;
+    let package = repo_root.target.generated_java.join_iter(GENERATED_CODE_NAMESPACE);
+    let test = package.join(GENERATED_TEST_CLASS).with_extension(JAVA_EXTENSION);
+    let mut test_class = GENERATED_CODE_NAMESPACE.into_iter().chain(Some(GENERATED_TEST_CLASS));
```

Comment:
Does this need to be `mut`?

Reply:
It does because I'm passing the value to `itertools` `join`, which takes `&mut`--however I agree the `mut` looks weird, so I refactored it.